webui/react/src/pages/ExperimentList.tsx

```diff
@@ -868,6 +854,27 @@ const ExperimentList: React.FC<Props> = ({ project }) => {
     return () => currentCanceler.abort();
   }, []);
 
+  // Dynamically change the confirmation modal config based on the selected batch action.
+  const confirmModalOptions = useMemo(() => {
+    return batchAction ? {
+      content: (
+        <>
+          Are you sure you want to <b>{batchAction.toLocaleLowerCase()}</b> all the eligible selected experiments?
+        </>
+      ),
+      danger: /(cancel|kill)/i.test(batchAction),
+      okText: /cancel/i.test(batchAction) ? 'Confirm' : batchAction,
+      onClose: () => setBatchAction(undefined),
+      onOk: () => submitBatchAction(batchAction),
+      title: `Confirm Batch ${batchAction}`,
+    } : { content: '', title: '' };
+  }, [batchAction, submitBatchAction]);
+
+  // Open the batch confirmation modal when confirm modal options has a title set.
+  useEffect(() => {
+    if (confirmModalOptions.title) ConfirmModal.open();
+  }, [ConfirmModal, confirmModalOptions.title]);
+
```

Comment:
I don't think we need the props to be dynamically defined, or to call `open()` in a `useEffect`. We can just set the `batchAction` state value and `open()` in `handleBatchAction`